### PR TITLE
Feat/met 666 create library for cleaning and normalization

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   * @return The action.
+   * @throws NormalizationConfigurationException In case there was an issue configuring the action.
+   */
+  NormalizeAction createAction(NormalizerSettings settings)
+      throws NormalizationConfigurationException {
+    return this.actionCreator.create(settings);
+  }
+
+  @FunctionalInterface
+  private static interface ActionCreator {
```
Comment (stzanakis) [inline metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java]: static redundant

File: metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public NormalizeAction create(NormalizerSettings settings)
```
Comment (stzanakis) [inline metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java]: public redundant

File: metis-normalization/.gitignore [hunk 3]
```diff
@@ -1,51 +1 @@
...
-
-# TeXlipse plugin
-.texlipse
-
-# STS (Spring Tool Suite)
-.springBeans
-
-# Code Recommenders
-.recommenders/
+target/
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/.gitignore]: File removed.

File: metis-normalization/src/main/java/eu/europeana/normalization/Normalizer.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  /**
+   * This method normalizes one EDM record (which is a string representing an EDM XML file),
+   * reporting on the process.
+   * 
+   * @param edmRecord The record to normalize.
+   * @return The normalized record, again as an EDM XML file, along with a report on the
+   *         normalization.
+   * @throws NormalizationException In case there was a problem.
+   */
+  public NormalizationResult normalize(String edmRecord) throws NormalizationException;
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/Normalizer.java]: Removed.

File: metis-normalization/src/main/java/eu/europeana/normalization/Normalizer.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  /**
+   * This method is a convenience method for calling {@link #normalize(String)} for multiple EDM
+   * records.
+   * 
+   * @param edmRecords The records to normalize.
+   * @return The normalized records, long with reports on the normalizations.
+   * @throws NormalizationException In case there was a problem.
+   */
+  public NormalizationBatchResult normalize(List<String> edmRecords) throws NormalizationException;
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/Normalizer.java]: Removed.

File: metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java]: Removed.

File: metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  public List<LanguageLabel> getLabels() {
+    return Collections.unmodifiableList(labels);
+  }
+
+  /**
+   * @return A set containing all labels that are known for this language, including the original
+   *         and alternative names.
+   */
+  public Set<String> getAllLabels() {
+    return Arrays.asList(originalNames, alternativeNames, labels).stream().flatMap(List::stream)
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java]: Changed.

File: metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  /**
+   * Get the preferred label of this language object that matches the given result language. For
+   * instance, this may be requesting the preferred name for the English language in Dutch.
+   * 
+   * @param resultLanguageCode The result language.
+   * @return The label.
+   */
+  public String getPrefLabel(String resultLanguageCode) {
+    final LanguageLabel label =
+        Arrays.asList(originalNames, alternativeNames, labels).stream().flatMap(List::stream)
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java]: Changed.

File: metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    return Arrays.asList(originalNames, alternativeNames, labels).stream().flatMap(List::stream)
+        .map(LanguageLabel::getLabel).collect(Collectors.toSet());
+  }
+
+  /**
+   * @return A set containing all labels and codes that are known for this language.
+   */
+  public Set<String> getAllLabelsAndCodes() {
+    final Set<String> result = getAllLabels();
+    final Set<String> codes = Arrays.asList(iso6391, iso6392b, iso6392t, iso6393, authorityCode)
```
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/languages/Language.java]: Changed.

File: metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java (hunk 2 above)
Comment (jochen-vermeulen) [reply metis-normalization/src/main/java/eu/europeana/normalization/NormalizerStep.java]: Removed.

**Pull request merged**